### ISSUE TITLE
The plugin crashes unexpectedly when using `...rest` syntax in the destructuring of the result of calling useForm

### ISSUE THREAD

**Issue by akd-io** [andykao1213/eslint-plugin-react-hook-form#14]
**Describe the bug**
The plugin crashes unexpectedly when using `...rest` syntax in the destructuring of the result of calling useForm.

**To Reproduce**
1. The code you are going to lint.
```jsx
const { useForm } = require('react-hook-form')
export function Component() {
  const { watch, ...restFormMethods } = useForm()
  return null
}
```
2. Your ESlint config
```
{
  ...
  'react-hook-form/no-nested-object-setvalue': ['error', { bracketAsArrayIndex: true }],
}
```

**Expected behavior**
I expect the React Hook Form Eslint plugin to not crash.

**Actual behavior**
The React Hook Form ESLing plugin crashes with the following error message:
```
TypeError: Cannot read property 'name' of undefined
Occurred while linting my-project/components/TestComponent.jsx:3
    at my-project/node_modules/eslint-plugin-react-hook-form/lib/rules/no-nested-object-setvalue.js:49:24
    at Array.find (<anonymous>)
    at check (my-project/node_modules/eslint-plugin-react-hook-form/lib/rules/no-nested-object-setvalue.js:48:53)
    at my-project/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (my-project/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (my-project/node_modules/eslint/lib/linter/node-event-generator.js:254:26)
    at NodeEventGenerator.applySelectors (my-project/node_modules/eslint/lib/linter/node-event-generator.js:283:22)
    at NodeEventGenerator.enterNode (my-project/node_modules/eslint/lib/linter/node-event-generator.js:297:14)
    at CodePathAnalyzer.enterNode (my-project/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
```

**Enviornment (please complete the following information):**
 - Eslint version: 7.12.1
 - Node version: v14.18.3
 - React-Hook-Form version: 6.15.8
 - eslint-plugin-react-hook-form version: 0.2.2

**Additional context**
n/a

**Comment by andykao1213** [comment]
Fixed in PR #15

**Comment by andykao1213** [comment]
@akd-io The issue has been fixed in [v0.2.3](https://github.com/andykao1213/eslint-plugin-react-hook-form/releases/tag/v0.2.3). Please if this version works for you, thx.

**Comment by akd-io** [comment]
Issue seems fixed and ESLint runs without problems, thanks! 😊 Now facing minor issue #17.
